errai-bus/src/main/java/org/jboss/errai/bus/server/service/ErraiServiceProxy.java

```diff
@@ -83,8 +86,10 @@ public void setDispatcher(RequestDispatcher dispatcher) {
     throw new IllegalStateException("cannot set dispatcher in proxy");
   }
 
-  public boolean isClosed() {
-    return service != null;
+  public void reset() {
+    service = null;
+    messageBusProxy = new MessageBusProxy();
```

Comment:
I don't think replacing the bus and dispatcher proxies is a good idea. There could be references to the old proxies that are held by other classes. Like this proxy, you will probably want to reset these bus/dispatcher proxies and close the same instances again.

Reply:
Okay